### PR TITLE
Regridding with CDO

### PR DESCRIPTION
**Pull request by s-scherrer**
This adds the option to regrid data directly after downloading it using CDO. The regridding is done using pre-computed weights in a separate thread in order to not block the download.

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -33,9 +33,9 @@ def test_load_var_table():
     assert table.loc[100].short_name == 'msdrswrf'
 
     table = load_var_table('era5-land')
-    assert table.index.size == 50
-    assert table.loc[46].dl_name == 'volumetric_soil_water_layer_1'
-    assert table.loc[46].short_name == 'swvl1'
+    assert table.index.size == 49
+    assert table.loc[45].dl_name == 'volumetric_soil_water_layer_1'
+    assert table.loc[45].short_name == 'swvl1'
```
Comment (s-scherrer) [inline tests/test_utils.py:38]: I had to adapt this test, but I think this is unrelated to my changes.

Comment (wpreimes) [reply tests/test_utils.py:38]: yeah, I think I deleted one variable some time ago from the table because it is not available anymore
